src/compile/data/assemble.ts

```diff
@@ -36,6 +42,79 @@ function removeUnnecessaryNodes(node: DataFlowNode) {
 
   node.children.forEach(removeUnnecessaryNodes);
 }
+
+/**
+ * Clones the subtree and ignores output nodes except for the leafs, which are renamed.
+ */
+function cloneSubtree(facet: FacetNode) {
+  function clone(node: DataFlowNode): DataFlowNode[] {
+    if (isClonable(node)) {
+      const copy = node.clone();
+
+      if (copy instanceof OutputNode) {
+        const newName = FACET_SCALE_PREFIX + facet.model.getName(copy.source);
+        copy.source = newName;
+
+        facet.model.component.data.outputNodes[newName] = copy;
+
+        flatten(node.children.map(clone)).forEach((n: DataFlowNode) => n.parent = copy);
+      } else if (copy instanceof AggregateNode) {
+        copy.addDimensions(facet.fields);
+
+        flatten(node.children.map(clone)).forEach((n: DataFlowNode) => n.parent = copy);
+      } else if (copy instanceof BinNode || node instanceof NonPositiveFilterNode || node instanceof NullFilterNode || node instanceof OrderNode || node instanceof StackNode || node instanceof TimeUnitNode || node instanceof FilterNode || node instanceof CalculateNode) {
+        flatten(node.children.map(clone)).forEach((n: DataFlowNode) => n.parent = copy);
+      }
+
+      return [copy];
+    }
+
+    return flatten(node.children.map(clone));
+  }
+  return clone;
+}
+
+/**
+ * Move facet nodes down to the next fork or output node. Also pull the main output with the facet node.
+ * After moving down the facet node, make a copy of the subtree and make it a child of the main output.
+ */
+function moveFacetDown(node: DataFlowNode) {
+  if (node instanceof FacetNode) {
+    if (node.numChildren() === 1 && !(node.children[0] instanceof OutputNode)) {
```

Comment:
I think if there is no fork but there is a unit plot with some shared scales and some independent scales, we have to perform a similar subtree clone procedure.  

If you think this is true, we can file an issue and handle this later though.